primitives/src/asset.rs

```diff
@@ -16,12 +16,39 @@
 #![allow(missing_docs)]
 
 use crate::IdentityId;
+use base64;
 use codec::{Decode, Encode};
+use frame_support::dispatch::DispatchError;
 use polymesh_primitives_derive::VecU8StrongTyped;
 #[cfg(feature = "std")]
 use serde::{Deserialize, Serialize};
 use sp_std::prelude::*;
 
+#[derive(
+    Encode, Decode, Clone, Debug, PartialEq, Eq, VecU8StrongTyped, Default, PartialOrd, Ord,
+)]
+pub struct Base64Vec(pub Vec<u8>);
```

Comment:
I think it's better to define
* `impl TryFrom<AsRef<[u8]>> for Base64Vec` that does `encode` and
* `impl From<Base64Vec> for AsRef<[u8]>` that does `decode`
and do not derive `VecU8StrongTyped` which would not allow the above instances to be defined. These would be essentially replace the `Base64Vec` methods `new` and `decode`. See `primitives_derive/src/vec_u8_strong_typed.rs` for the starting point for those instances.

Then conversion is done by `.try_into()?` and `.into()`.

Reply:
An even better solution would be to define a derive macro, say `VecU8Base64StrongTyped`, that can be derived for `EncryptionPubKeyWrapper` instead of `VecU8StrongTyped` but has a `TryFrom` instance (failing on vectors that are not base-64 encodings) instead of the `From` instance.

Reply:
Added it to https://polymath.atlassian.net/browse/MESH-1331

Merging, since @maxsam4 believed that we don't need the conversion to base64 at all and we can instead upload a file with binary data. Therefore, let's first make sure that we need it, before making it nicer....